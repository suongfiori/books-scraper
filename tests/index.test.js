const request = require('supertest')
const app = require('../src/index')
const { getGenreParam } = require('../src/paramGenerator') 

describe('GET /books', () => {
  beforeAll(async () => {
    await getGenreParam()
  }) 

  it('responds with an array of genres', async () => {
    const response = await request(app).get('/books')

    expect(response.status).toBe(200)
    expect(Array.isArray(response.body)).toBe(true)
    expect(response.body.length).toBeGreaterThan(0)
    expect(response.body).toMatchSnapshot()
  })
})

describe('GET /books/:genre', () => {
  it('responds with JSON containing scraped book data', async () => {
  
    const response = await request(app).get('/books/childrens').set('Accept', 'application/json');

    expect(response.status).toBe(200);
    expect(response.type).toBe('application/json')
    expect(response.body).toHaveProperty('status', 'success')
    expect(response.body).toHaveProperty('data')
    expect(response.body).toHaveProperty('metadata')
    expect(response.body.metadata).toHaveProperty('genre', 'childrens')
    expect(response.body.metadata).toHaveProperty('total_items')
  })

  it('responds with 400 for invalid genre parameter', async () => {
    const response = await request(app).get('/books/invalidGenre')

    expect(response.status).toBe(400)
    expect(response.type).toBe('application/json')
    expect(response.body).toHaveProperty('error', 'Invalid genre parameter.')
  })

})

