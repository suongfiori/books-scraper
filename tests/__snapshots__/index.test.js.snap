// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /books responds with an array of genres 1`] = `
[
  "travel",
  "mystery",
  "historical fiction",
  "sequential art",
  "classics",
  "philosophy",
  "romance",
  "womens fiction",
  "fiction",
  "childrens",
  "religion",
  "nonfiction",
  "music",
  "default",
  "science fiction",
  "sports and games",
  "add a comment",
  "fantasy",
  "new adult",
  "young adult",
  "science",
  "poetry",
  "paranormal",
  "art",
  "psychology",
  "autobiography",
  "parenting",
  "adult fiction",
  "humor",
  "horror",
  "history",
  "food and drink",
  "christian fiction",
  "business",
  "biography",
  "thriller",
  "contemporary",
  "spirituality",
  "academic",
  "self help",
  "historical",
  "christian",
  "suspense",
  "short stories",
  "novels",
  "health",
  "politics",
  "cultural",
  "erotica",
  "crime",
]
`;
